### ISSUE TITLE
Improve `log.error()` signature

### ISSUE THREAD

**Issue by unional** [aurelia/logging#45]
version: 1.3.1

Improve `log.error(message: string, ...rest: any[])` to `log.error(messageOrError: string | Error, ...rest: any[])`

It is quite common for user to do:

```ts
try {
  doSomething()
}
catch (e) {
  log.error(e)
}
```

**Comment by EisenbergEffect** [comment]
Sounds like a great opportunity to contribute a PR :)

**Comment by Alexander-Taran** [comment]
same solution
log.info(`${e}`)

can be closed

**Comment by EisenbergEffect** [comment]
Not sure it will do exactly the same thing.

**Comment by Alexander-Taran** [comment]
Yeah.. not the same.. 
will be toString()

log.error(`${e}`,e) on other hand..

**Comment by Alexander-Taran** [comment]
won't fix then (-:
or keep in mind?

**Comment by unional** [comment]
IMO it should be fixed, the `log.?()` interface should match with `console.?()` which accepts anything.

PR was created and closed because it can be a breaking change.

Should consider fixing this when a major release is pending.

**Comment by Alexander-Taran** [comment]
well.. console could be not the only appender
and human readable message is knida the point..

**Comment by unional** [comment]
Separation of concern. The appender should handle the formatting, not the logger.

**Comment by Alexander-Taran** [comment]
Depends
Had a presentation recently about logging @ workplace

With serilog
You actually might want to have that message defined after all and not lazy-pass exception to logger

If you talk about parsing the production logs that is
Not when you are debugging and just need to dump something to console temporarily

**Comment by unional** [comment]
I agree on that argument.  However, if you take about temporary logs, you may as well just use `console.debug()`. 

Using Aurelia logging should hear more toward production worthy logs. As only those logs would benefit from the runtime log level manipulation and multiple appenders support. 

For example, temporary logs don’t need the capability of the `aurelia-logging-color` that I wrote.

**Comment by EisenbergEffect** [comment]
Happy to take a PR...
